packages/weather/OpenWeatherMap.py

```diff
@@ -43,32 +44,103 @@ def wrapper_load_config(string, entities):
         return func(payload)
     return wrapper_load_config
 
+
 def acquire_weather(func):
     @functools.wraps(func)
     def wrapper_acquire_weather(payload):
         for item in payload["entities"]:
-            if item["entity"] == "city":
+            if item["entity"] == "date" or item["entity"] == "datetime":
+                payload["time_flag"] = 1
+                return func(payload)
+            elif item["entity"] == "city":
                 utils.output("inter", "acquiring", utils.translate("acquiring"))
 
                 payload["city"] = item["sourceText"].title()
-                payload["observation"] = payload["owm"].weather_at_place(payload["city"])
+                try:
+                    payload["observation"] = payload["owm"].weather_at_place(payload["city"])
+                except:
+                    tmp = payload["city"]
+
+                    es = tmp.find(' ')
+                    vi = tmp.find(',')
+                    pt = tmp.find('.')
+
+                    if es != -1:
+                        tmp = tmp[:es]
+                    elif vi != -1:
+                        tmp = tmp[:es]
+                    elif pt != -1:
+                        tmp = tmp[:es]
+                    payload["city"] = tmp
+                    try:
+                        payload["observation"] = payload["owm"].weather_at_place(payload["city"])
+                    except:
+                        return utils.output("end", "not_found", utils.translate("not_found"))
                 payload["wtr"] = payload["observation"].get_weather()
 
                 return func(payload)
+
         return utils.output("end", "request_error", utils.translate("request_error"))
 
     return wrapper_acquire_weather
 
 
+def future(func):
```

Comment:
Since the decorator ffor present weather is called `acquire_weather` and since you called the wrapper for the future weather `wrapper_acquire_future_weather` I'd call this decorator acquire_future_weather` for consistency.

Reply:
I rename it.